libcalico-go/lib/backend/k8s/resources/workloadendpoint.go

```diff
@@ -53,6 +53,27 @@ type WorkloadEndpointClient struct {
 	converter conversion.Converter
 }
 
+type patchModeKey struct{}
+type PatchMode string
+
+const (
+	PatchModeCNI           PatchMode = "patchCNI"
+	PatchModeEgressGateway PatchMode = "patchEgressGateway"
+	PatchModeUnspecified   PatchMode = "patchUnspecified"
```

Comment:
all of the values here should probably be `patchModeX`, else logs will say things like "patchUnspecified" which suggests that the patch contents are unspecified rather than the patch mode not being provided.

Reply:
Great point! I'll change them.